auth/provider.go

```diff
@@ -114,6 +122,29 @@ func (r *Keycloak) newScope(s string) (scope KeycloakScope) {
 	return
 }
 
+//
+// Username resolves token to username using Keycloak service.
+func (r *Keycloak) Username(token string) (username string, err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+	defer cancel()
+	decoded, claims, err := r.client.DecodeAccessToken(ctx, token, r.realm)
```

Comment:
Since the middleware has already authenticated the token - how can this timeout? fail?

Reply:
It looked to me that once it is a method exposed from the auth package, it should be self-sufficient with error checking, but I agree there is the middeware method being part of the auth pkg which forces the Scopes check - will update lines below keeping just parsing of the "preffered_username" which is not touched by the Scopes method and it depends on Keycloak service config.

Reply:
Agreed the method should be self sufficient.  Mainly wanted to understand how it could fail.